fbw-common/src/wasm/systems/systems/src/hydraulic/command_sensor_unit.rs

```diff
@@ -0,0 +1,336 @@
+            value,
+            CSU::Conf0 | CSU::Conf1 | CSU::Conf2 | CSU::Conf3 | CSU::ConfFull
+        )
+    }
+
+    // The CSU is made of two parallel switches: this is why there is `switch_pattern_1` and
+    // `switch_pattern_2`.
+    pub fn get_csu_configuration(
+        switch_pattern_1: SwitchPattern,
+        switch_pattern_2: SwitchPattern,
+    ) -> CSU {
+        match (switch_pattern_1, switch_pattern_2) {
+            (0b11000, 0b11000) => CSU::Conf0,
+            (0b11000, 0b01000) => CSU::OutOfDetent,
+            (0b01000, 0b01000) => CSU::OutOfDetent,
+            (0b01000, 0b01100) => CSU::OutOfDetent,
+            (0b01000, 0b11000) => CSU::OutOfDetent,
+            (0b01100, 0b01000) => CSU::OutOfDetent,
+            (0b01100, 0b01100) => CSU::Conf1,
+            (0b01100, 0b00100) => CSU::OutOfDetent,
+            (0b00100, 0b00100) => CSU::OutOfDetent,
+            (0b00100, 0b00110) => CSU::OutOfDetent,
+            (0b00100, 0b01100) => CSU::OutOfDetent,
+            (0b00110, 0b00100) => CSU::OutOfDetent,
+            (0b00110, 0b00110) => CSU::Conf2,
+            (0b00110, 0b00010) => CSU::OutOfDetent,
+            (0b00010, 0b00010) => CSU::OutOfDetent,
+            (0b00010, 0b00011) => CSU::OutOfDetent,
+            (0b00010, 0b00110) => CSU::OutOfDetent,
+            (0b00011, 0b00010) => CSU::OutOfDetent,
+            (0b00011, 0b00011) => CSU::Conf3,
+            (0b00011, 0b00001) => CSU::OutOfDetent,
+            (0b00001, 0b00001) => CSU::OutOfDetent,
+            (0b00001, 0b10001) => CSU::OutOfDetent,
+            (0b10001, 0b00001) => CSU::OutOfDetent,
+            (0b00001, 0b00011) => CSU::OutOfDetent,
+            (0b10001, 0b10001) => CSU::ConfFull,
+            _ => CSU::Fault,
+        }
+    }
+}
+
+pub struct CSUMonitor {
+    handle_position_id: VariableIdentifier,
+
+    current_switch_pattern: (SwitchPattern, SwitchPattern),
+    current_detent: CSU,
+    previous_switch_pattern: (SwitchPattern, SwitchPattern),
+    previous_detent: CSU,
+
+    last_valid_switch_pattern: (SwitchPattern, SwitchPattern),
+    time_since_last_valid_detent: Duration,
+}
+impl CSUMonitor {
+    pub fn new(context: &mut InitContext) -> Self {
+        Self {
+            handle_position_id: context.get_identifier("FLAPS_HANDLE_INDEX".to_owned()),
+            current_switch_pattern: (0b11000, 0b11000),
+            current_detent: CSU::Conf0,
+            previous_switch_pattern: (0b11000, 0b11000),
+            previous_detent: CSU::Conf0,
+            last_valid_switch_pattern: (0b11000, 0b11000),
+            time_since_last_valid_detent: Duration::ZERO,
+        }
+    }
+
+    pub fn update(&mut self, context: &UpdateContext) {
+        self.time_since_last_valid_detent += context.delta();
+    }
+
+    fn get_csu_configuration(&self) -> CSU {
+        CSU::get_csu_configuration(self.current_switch_pattern.0, self.current_switch_pattern.1)
+    }
+
+    pub fn get_last_valid_detent(&self) -> CSU {
```

Comment:
```suggestion
    #[cfg(test)]
    fn get_last_valid_detent(&self) -> CSU {
```
No unnecessary `pub` function.